src/main/java/com/hotels/hcommon/ssh/session/DelegatingSessionFactoryBuilder.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (C) 2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.hcommon.ssh.session;
+
+import java.util.Map;
+
+import com.jcraft.jsch.Proxy;
+import com.pastdev.jsch.SessionFactory;
+import com.pastdev.jsch.SessionFactory.SessionFactoryBuilder;
+
+class DelegatingSessionFactoryBuilder extends SessionFactoryBuilder {
+  final /* VisibleForTesting */ SessionFactoryBuilder delegate;
```

Comment:
Same comments here as for class above

Reply:
same as above, the timeout can be private

Reply:
OK, just move comment